templates/vue-ts/src/Feedback.vue

```diff
@@ -36,29 +32,37 @@ const { snapshot, send } = useMachine(feedbackMachine, {
       </button>
     </div>
 
-
-
     <div v-if="snapshot.matches('thanks')" class="step">
       <h2>Thanks for your feedback.</h2>
 
-      <p v-if="snapshot.context.feedback.length > 0">"{{ snapshot.context.feedback }}"</p>
-
+      <p v-if="snapshot.context.feedback.length > 0">
+        "{{ snapshot.context.feedback }}"
+      </p>
     </div>
 
-
-    <form v-if="snapshot.matches('form')" class="step" @submit.prevent="send({ type: 'submit' })">
+    <form
+      v-if="snapshot.matches('form')"
+      class="step"
+      @submit.prevent="send({ type: 'submit' })"
+    >
       <h2>What can we do better?</h2>
-      <textarea name="feedback" rows={4} placeholder="So many things..." @input="(ev) => {
-        send({
-          type: 'feedback.update',
-          value: (ev.target as HTMLTextAreaElement).value
-        })
-      }" />
+      <textarea
+        name="feedback"
+        rows="{4}"
```

Comment:
```suggestion
        rows="4"
```

Reply:
I was getting a type error locally on that about it needing to be a number but I realized it's because it was loading react types for vue files? No idea what was going on there

Reply:
Might have been my fault

Reply:
> I was getting a type error locally on that about it needing to be a number but I realized it's because it was loading react types for vue files? No idea what was going on there

Vue relies on the global JSX namespace so it tends to conflict with React which still has that global namespace too (no more in the upcoming React 19!). We have a patch meant to fight it [here](https://github.com/statelyai/xstate/blob/71183da18ae6b0b6b2e8f0c52ea9976232e54f41/patches/%40vue%2Bruntime-dom%2B3.0.11.patch) but I'm not sure what's the current status of this thing. It could easily stop working at some point